From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pierre Ossman <ossman@cendio.se>
Date: Mon, 22 Nov 2021 13:53:05 +0100
Subject: [PATCH] Ignore ResizeObserver errors

It seems that Firefox has a bug where these are fired incorrectly when
we are in an <iframe>. The events also contain no useful details, so we
can't really do anything useful with them anyway.

Signed-off-by: Dominik Csapak <d.csapak@proxmox.com>
---
 app/error-handler.js | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/app/error-handler.js b/app/error-handler.js
index 81a6cba..f956c22 100644
--- a/app/error-handler.js
+++ b/app/error-handler.js
@@ -21,6 +21,12 @@
         try {
             const msg = document.getElementById('noVNC_fallback_errormsg');
 
+            // Work around Firefox bug:
+            // https://bugzilla.mozilla.org/show_bug.cgi?id=1685038
+            if (event.message === "ResizeObserver loop completed with undelivered notifications.") {
+                return false;
+            }
+
             // Only show the initial error
             if (msg.hasChildNodes()) {
                 return false;
